extension/observer/kafkatopicsobserver/config.go

```diff
@@ -23,11 +24,30 @@ type Config struct {
 	// Session interval for the Kafka consumer
 	SessionTimeout time.Duration `mapstructure:"session_timeout"`
 	// Heartbeat interval for the Kafka consumer
-	HeartbeatInterval time.Duration        `mapstructure:"heartbeat_interval"`
-	Authentication    kafka.Authentication `mapstructure:"auth"`
-	TopicRegex        string               `mapstructure:"topic_regex"`
+	HeartbeatInterval  time.Duration        `mapstructure:"heartbeat_interval"`
+	Authentication     kafka.Authentication `mapstructure:"auth"`
+	TopicRegex         string               `mapstructure:"topic_regex"`
+	TopicsSyncInterval time.Duration        `mapstructure:"topics_sync_interval"`
 }
 
-func (config Config) Validate() error {
+func (config *Config) Validate() error {
+	if len(config.Brokers) == 0 {
+		return fmt.Errorf("brokers list must be specified")
+	}
+	if len(config.ProtocolVersion) == 0 {
+		return fmt.Errorf("protocol_version must be specified")
+	}
+	if len(config.TopicRegex) == 0 {
+		return fmt.Errorf("topic_regex must be specified")
+	}
+	if config.TopicsSyncInterval <= 0 {
+		return fmt.Errorf("topics_sync_interval must be greater than 0")
+	}
+	if config.SessionTimeout <= 0 {
+		return fmt.Errorf("session_timeout must be greater than 0")
+	}
+	if config.HeartbeatInterval <= 0 {
+		return fmt.Errorf("heartbeat_interval must be greater than 0")
+	}
 	return nil
 }
```

Comment:
This is my personal take (mostly as a dyslexic end user) but I prefer to know all the errors on the first verify so I can fix each up on one take.

Can I suggest that you do the following:

```suggestion
func (config *Config) Validate() (errs error) {
	if len(config.Brokers) == 0 {
		errs = multierr.Append(errs,fmt.Errorf("brokers list must be specified"))
	}
	if len(config.ProtocolVersion) == 0 {
		errs = multierr.Append(errs, fmt.Errorf("protocol_version must be specified"))
	}
	if len(config.TopicRegex) == 0 {
		errs = multierr.Append(errs,  fmt.Errorf("topic_regex must be specified"))
	}
	if config.TopicsSyncInterval <= 0 {
		errs = multierr.Append(errs,  fmt.Errorf("topics_sync_interval must be greater than 0"))
	}
	if config.SessionTimeout <= 0 {
		errs = multierr.Append(errs, fmt.Errorf("session_timeout must be greater than 0"))
	}
	if config.HeartbeatInterval <= 0 {
		errs = multierr.Append(errs,  fmt.Errorf("heartbeat_interval must be greater than 0"))
	}
	return errs
}
```

Reply:
You'll need to import the library as well and potentially format it.

Reply:
Done